src/TRoom.cpp

```diff
@@ -581,43 +581,36 @@ void TRoom::removeAllSpecialExitsToRoom(const int roomId)
 
 void TRoom::calcRoomDimensions()
 {
+    min_x = x;
+    max_x = x;
+    min_y = y;
+    max_y = y;
+
     if (customLines.empty()) {
         return;
     }
-    min_x = 0.0;
-    min_y = 0.0;
-    max_x = 0.0;
-    max_y = 0.0;
-    bool needInit = true;
 
     QMapIterator<QString, QList<QPointF>> it(customLines);
     while (it.hasNext()) {
         it.next();
-        const QList<QPointF>& _pL = it.value();
-        if (_pL.empty()) {
+        const QList<QPointF>& pointsInLine = it.value();
+        if (pointsInLine.empty()) {
             continue;
         }
-        if (needInit) {
-            needInit = false;
-            min_x = _pL[0].x();
-            max_x = min_x;
-            min_y = _pL[0].y();
-            max_y = min_y;
-        }
-        for (auto point : _pL) {
-            qreal _x = point.x();
-            qreal _y = point.y();
-            if (_x < min_x) {
-                min_x = _x;
+        for (auto pointInLine : pointsInLine) {
```

Comment:
I know the old code did this - but when you use Qt containers their copy-on-write behaviour can trigger accidentally and make a copy list - doesn't hurt the data but hurts performance. Have to wrap the access in a [qAsConst()](https://doc.qt.io/qt-5/qtglobal.html#qAsConst). `std::` ones don't suffer from this btw
```suggestion
        for (auto pointInLine : qAsConst(pointsInLine)) {
```

Reply:
Um, `pointsInLine` ***is*** a `const` reference already and it was obtained from a `QMap<QString, QList<QPointF>>::value()` which is, IIRC a copy of the original data - so Qt's CoW actions cannot be triggered here AFAICT.

Reply:
IMVHO If you feel strongly enough about this then we can change this in another PR.